### PR TITLE
superTuxKart: fix missing dependency on darwin

### PR DESCRIPTION
**Pull request by nrhtr**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by bjornfor — APPROVED** [review]

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -37754,7 +37754,7 @@ with pkgs;
   superTux = callPackage ../games/supertux { };
 
   superTuxKart = callPackage ../games/super-tux-kart {
-    inherit (darwin.apple_sdk.frameworks) Cocoa IOKit OpenAL;
+    inherit (darwin.apple_sdk.frameworks) Cocoa IOKit OpenAL IOBluetooth;
```
Comment (OPNA2608) [inline pkgs/top-level/all-packages.nix:37757]:
```suggestion
    inherit (darwin.apple_sdk_11_0.frameworks) Cocoa IOKit OpenAL IOBluetooth;
```

**Review by OPNA2608 — CHANGES_REQUESTED** [review]
Needs newer SDK for x86_64-darwin.
```
In file included from /tmp/nix-build-supertuxkart-1.4.drv-0/source/lib/graphics_engine/src/ge_compressor_astc_4x4.cpp:4:
In file included from /tmp/nix-build-supertuxkart-1.4.drv-0/source/lib/graphics_engine/src/ge_vulkan_command_loader.hpp:4:
In file included from /tmp/nix-build-supertuxkart-1.4.drv-0/source/lib/graphics_engine/include/vulkan_wrapper.h:13:
/tmp/nix-build-supertuxkart-1.4.drv-0/source/lib/graphics_engine/include/vk_mvk_moltenvk.h:29:10: fatal error: 'IOSurface/IOSurfaceRef.h' file not found
#include <IOSurface/IOSurfaceRef.h>
         ^~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/nix-build-supertuxkart-1.4.drv-0/source/lib/graphics_engine/include/vk_mvk_moltenvk.h:29:10: note: did not find header 'IOSurfaceRef.h' in framework 'IOSurface' (loaded from '/nix/store/aqv6fxaj2cmv14xkhh1vqj0qzcch32v2-apple-framework-IOSurface/Library/Frameworks')
```

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -37754,7 +37754,7 @@ with pkgs;
   superTux = callPackage ../games/supertux { };
 
   superTuxKart = callPackage ../games/super-tux-kart {
```
[the remark this answers is not in the document]
Comment (nrhtr) [reply pkgs/top-level/all-packages.nix:37756]: Sounds good to me!

**Review by OPNA2608 — APPROVED** [review]
Works for me on x86_64-darwin.

**Pull request merged**